server/src/model.ts

```diff
@@ -1019,6 +1019,55 @@ export async function getAssignmentStudentData({
   return { ...assignmentData, documentScores };
 }
 
+export async function getAllAssignmentStudentData({
+  userId,
+} : {
+  userId: number;
+}) {
+
+  const userScores = await prisma.users.findMany({
+    where: { 
+      assignmentScores: {
+        some: {
+          assignment: {
+            ownerId: userId
+          }
+        }
+      }
+    },
+    include: {
+      assignmentScores: {
+        where: {
+          assignment: {
+            ownerId: userId
+          }
+        },
+        select: {
+          assignmentId: true,
+          userId: true,
+          score: true
+        },
+      },
+    },
+  });
+
+  const assignments = await prisma.assignments.findMany({
```

Comment:
Can you do this all in a single query? For example, if you `include` assignmentScores from inside this `assignments` query, does prisma expand it to include the scores of all students? Or, you could start with the `assignmentScores` table and then add what you need from the `assignments` and `users` table via includes. I'm pretty sure this second suggestion would work, but not so sure about the first one.

Reply:
The reason I did it this way was because I wanted to have the top level of keys in the dictionary be students and not assignments. The way AssignmentData, which I used for reference, populated the chart was going row by row, and I was having trouble coming up with an equally clean way to populate the table without jumping around filling things in. Maybe we could talk about this next time we meet?

Reply:
Yeah, I try to minimize the number of database queries, but then sometimes end up with two or more for reasons like you mention.

If you queried based on `assignmentScores`, you'd get multiple rows per students; I assume one row per assignment that they took. If you sorted it by student and then assignment, it would be in row-major order like you want, but the problem is that each row would be missing assignments that the student didn't take and you wouldn't know what all the columns would be... You'd still need a second query to get all the assignments, so this might not be an improvement.

We can discuss this next time, though I'm not optimistic we'll be able to improve what you have already.

Reply:
Charles and I thought we had come up with a one-query solution last night, but then we realized it didn't include assignments that no one had done yet